examples/mnist/mnist_deep_simplified_distributed.py

```diff
@@ -0,0 +1,272 @@
+    # is down to 7x7x64 feature maps -- maps this to 1024 features.
+    with tf.name_scope('fc1'):
+        W_fc1 = weight_variable([7 * 7 * 64, 1024], "W_fc1")
+        b_fc1 = bias_variable([1024])
+
+        h_pool2_flat = tf.reshape(h_pool2, [-1, 7 * 7 * 64])
+        h_fc1 = tf.nn.relu(tf.matmul(h_pool2_flat, W_fc1) + b_fc1)
+
+    # Map the 1024 features to 10 classes, one for each digit
+    with tf.name_scope('fc2'):
+        W_fc2 = weight_variable([1024, 10], "W_fc2")
+        b_fc2 = bias_variable([10])
+
+        # y_conv = tf.matmul(h_fc1_drop, W_fc2) + b_fc2
+        y_conv = tf.matmul(h_fc1, W_fc2) + b_fc2
+    return y_conv, tf.placeholder(tf.float32)
+
+
+def conv2d(x, W):
+    """conv2d returns a 2d convolution layer with full stride."""
+    return tf.nn.conv2d(x, W, strides=[1, 1, 1, 1], padding='SAME')
+
+
+def max_pool_2x2(x):
+    """max_pool_2x2 downsamples a feature map by 2X."""
+    return tf.nn.max_pool(
+        x, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='SAME')
+
+
+def weight_variable(shape, name):
+    """weight_variable generates a weight variable of a given shape."""
+    initial = tf.get_variable(name, shape)
+    return tf.Variable(initial)
+
+
+def bias_variable(shape):
+    """bias_variable generates a bias variable of a given shape."""
+    initial = tf.constant(0.1, shape=shape)
+    return tf.Variable(initial)
+
+
+def train_mnist_cnn(FLAGS):
+    # Config
+    config = tf.ConfigProto(
+        allow_soft_placement=True,
+        log_device_placement=False,
+        inter_op_parallelism_threads=1)
+
+    # Note: Additional configuration option to boost performance is to set the
+    # following environment for the run:
+    # OMP_NUM_THREADS=44 KMP_AFFINITY=granularity=fine,scatter
+    # The OMP_NUM_THREADS number should correspond to the number of
+    # cores in the system
+
+    # Import data
+    mnist = input_data.read_data_sets(FLAGS.data_dir, one_hot=True)
+
+    # Create the model
+    x = tf.placeholder(tf.float32, [None, 784])
+
+    # Define loss and optimizer
+    y_ = tf.placeholder(tf.float32, [None, 10])
+
+    # Build the graph for the deep net
+    y_conv, keep_prob = deepnn(x)
+
+    with tf.name_scope('loss'):
+        cross_entropy = tf.nn.softmax_cross_entropy_with_logits(
+            labels=y_, logits=y_conv)
+    cross_entropy = tf.reduce_mean(cross_entropy)
+
+    # add distributed wrapper to "adam_optimizer"
+    opt = hvd.DistributedOptimizer(tf.train.AdamOptimizer(1e-4))
+    global_step = tf.contrib.framework.get_or_create_global_step()
+    with tf.name_scope('distributed_optimizer'):
+        train_step = opt.minimize(cross_entropy, global_step=global_step)
+
+    with tf.name_scope('accuracy'):
+        correct_prediction = tf.equal(tf.argmax(y_conv, 1), tf.argmax(y_, 1))
+        correct_prediction = tf.cast(correct_prediction, tf.float32)
+    accuracy = tf.reduce_mean(correct_prediction)
+    tf.summary.scalar('Training accuracy', accuracy)
```

Comment:
I wonder if only rank/instance 0 should run the `tf.summary` or it doesn't matter?

Reply:
This does not seem matter. Checked the text output and tensorboard and they both looked fine.